src/Descriptor.ts

```diff
@@ -0,0 +1,62 @@
+enum DescriptorKind {
+  Package,
+  Type,
+  Term,
+  Meta,
+  Method,
+  Parameter,
+  TypeParameter,
+}
+
+export class Descriptor {
```

Comment:
Short doc strings for this type and the `syntax()` method? (I'm wondering if it would be more idiomatic to rename the `syntax()` method to `toString()`.)